models/models.py

```diff
@@ -32,10 +17,133 @@ def get_optimal_route(df, start_time, end_time, bar_num, total_max_walking_time,
     :param max_walking_each:
     :return: A Solution class object representing the optimal solution
     """
-    return
 
+    # parameters
+    bigm = 999999
+    m = Model("opt_route")
+    locations = df['name']
+    ratings = df['stars']
+    open_times = df['open']
+    close_times = df['close']
+    wait_times = df['wait_time'] / 60
+
+    time_spent_each_bar = max(0.25, (end_time - start_time - total_max_walking_time - max_total_wait) / bar_num)
+
+    y = []
+    x = [[0 for j in range(len(locations))] for i in range(len(locations))]
+    z = [[[0 for j in range(len(locations))] for i in range(len(locations))] for k in range(bar_num - 1)]
+    coordinates = list(zip(df.latitude, df.longitude))
+    dima = generate_distance_matrix(coordinates, df['business_id'], 'manhattan')
```

Comment:
We should only generate this distance matrix once outside of this function and feed it as an argument. That would be much faster. Also, do we use the distance CSV at all?